bokehjs/src/coffee/api/charts.coffee

```diff
@@ -0,0 +1,138 @@
+_ = require("underscore")
+$ = require("jquery")
+sprintf = require("sprintf")
+{Document} = require("../document")
+embed = require("../embed")
+models = require("./models")
+palettes = require("../palettes/palettes")
+
+sum = (array) ->
+  return array.reduce(((a, b) => a + b), 0)
+
+cumsum = (array) ->
+  result = []
+  array.reduce(((a, b, i) -> result[i] = a + b), 0)
+  return result
+
+num2hexcolor = (num) -> sprintf("#%06x", num)
+
+pie = (data, opts) ->
```

Comment:
JSDoc is the de facto standard: http://usejsdoc.org/

I'm sure we could hook into some output formatter to translate to RST/HTML/whatever for the doc site.

But I would say the first thing is JSDOC format for the API docs, just so people reading the source know how to call the function.